### ISSUE TITLE
Cannot return `std::vector<bool>`

### ISSUE THREAD

**Issue by TimPushkin** [scapix-com/scapix#43]
When I'm trying to generate an interface for a class with a method that returns `std::vector<bool>` I get the following error:
```
[100%] Building CXX object CMakeFiles/jnav.dir/generated/bridge/java/library.cpp.o
In file included from /mnt/d/Projects/jnav/generated/bridge/java/library.cpp:4:
In file included from /home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/bridge/java/init.h:10:
In file included from /home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/link/java/native_method.h:12:
/home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/link/java/convert.h:210:77: error: 'this' argument to member function 'data' has type 'const std::vector<bool, allocator<bool>>', but function is not marked const
                a->set_region(0, static_cast<jsize>(v.size()), reinterpret_cast<const J*>(v.data()));
                                                                                          ^
/home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/link/java/ref.h:163:60: note: in instantiation of member function 'scapix::link::java::convert<scapix::link::java::ref<scapix::link::java::array<unsigned char>, scapix::link::java::detail::api::scope::generic>, std::vector<bool>>::jni' requested here
        return convert<remove_cvref_t<Jni>, remove_cvref_t<Cpp>>::jni(std::forward<Cpp>(cpp));
                                                                  ^
/home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/link/java/native_method.h:137:29: note: in instantiation of member function 'scapix::link::java::param<scapix::link::java::ref<scapix::link::java::array<unsigned char>, scapix::link::java::detail::api::scope::generic>, std::vector<bool>>::jni' requested here
                                        return param<JniR, R>::jni((get_object<class_type>(thiz).*Method)(param<JniArgs, Args>::cpp(args)...));
                                                               ^
/home/eyjafl/.cmodule/scapix/1.0.30/source/scapix/link/java/native_method.h:186:168: note: in instantiation of member function 'scapix::link::java::native_method<scapix::meta::string<'t', 'e', 's', 't'>, scapix::link::java::ref<scapix::link::java::array<unsigned char>, scapix::link::java::detail::api::scope::generic> (scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>, scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>, scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>), std::vector<bool> (G2oTester::*)(std::vector<std::vector<double>>, std::vector<std::vector<double>>, std::vector<std::vector<double>>) const, &G2oTester::test>::type<true, scapix::link::java::ref<scapix::link::java::array<unsigned char>, scapix::link::java::detail::api::scope::generic> (scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>, scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>, scapix::link::java::ref<scapix::link::java::array<scapix::link::java::array<double>>, scapix::link::java::detail::api::scope::generic>), std::vector<bool> (std::vector<std::vector<double>>, std::vector<std::vector<double>>, std::vector<std::vector<double>>)>::func' requested here
        inline static constexpr auto func = &type<std::is_member_pointer_v<Type>, JniType, remove_function_qualifiers_t<member_pointer_type_t<std::remove_pointer_t<Type>>>>::func;
                                                                                                                                                                              ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/stl_bvector.h:964:7: note: 'data' declared here
      data() _GLIBCXX_NOEXCEPT { }
      ^
1 error generated.
```

Everything works if I just replace `bool` with `uint8_t`, for example.

**Comment by Boris-Rasin** [comment]
Thank you for reporting this issue.

I have committed the fix:
https://github.com/scapix-com/scapix/commit/3d39685848966f64f1015c90d1f6ec44096784c2

It will be available in the next release.

**Comment by Boris-Rasin** [comment]
Fix released in cmodule v1.0.36
https://www.scapix.com/language_bridge/changelog/
